pkg/errors/sdk_errors.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package errors
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+var (
+	ErrJSONKeyNotFound = errors.New("response body expected error message json key not found")
+	ErrUnknown         = errors.New("unknown error")
+)
+
+const err = "error"
+
+// SDKError is an error type for Mainflux SDK.
+type SDKError interface {
+	Error
+	StatusCode() int
+}
+
+var _ SDKError = (*sdkError)(nil)
+
+type sdkError struct {
+	*customError
+	statusCode int
+}
+
+func (ce *sdkError) Error() string {
+	if ce == nil {
+		return ""
+	}
+	if ce.customError == nil {
+		return http.StatusText(ce.statusCode)
+	}
+	return fmt.Sprintf("Status: %s: %s", http.StatusText(ce.statusCode), ce.customError.Error())
+}
+
+func (ce *sdkError) StatusCode() int {
+	return ce.statusCode
+}
+
+// NewSDKError returns an SDK Error that formats as the given text.
+func NewSDKError(err error) SDKError {
+	return &sdkError{
+		customError: &customError{
+			msg: err.Error(),
+			err: nil,
+		},
+		statusCode: 0,
+	}
+}
+
+// NewSDKErrorWithStatus returns an SDK Error setting the status code.
+func NewSDKErrorWithStatus(err error, statusCode int) SDKError {
+	return &sdkError{
+		statusCode: statusCode,
+		customError: &customError{
+			msg: err.Error(),
+			err: nil,
+		},
+	}
+}
+
+// CheckError will check for error in http response.
```

Comment:
Improve comment. Explain that in some cases (upsert), multiple status codes are valid.